experimental/experimental.go

```diff
@@ -2,3 +2,14 @@
 //
 // Note: All features here may be changed or deleted at any time, so use with caution!
 package experimental
+
+import "github.com/tetratelabs/wazero/api"
+
+// FunctionDefinition is an extend interface of api.FunctionDefinition with
+// additional experimental methods.
+type FunctionDefinition interface {
+	api.FunctionDefinition
+	// HumanName is a human-readable name for this function, which may not be
+	// the same as Name. It will fallback to DebugName() if not available.
+	HumanName() string
```

Comment:
What do you think about `SourceName` instead?

Reply:
Could you also speak of why not implement this in `api.FunctionDefinition.DebugName` directly?

Reply:
Happy with `SourceName`!

I didn't want to touch `DebugName` because it seemed like it has special semantic:

> DebugName identifies this function based on its Index or Name in the module.

For example, `HumanName` don't have any guarantees (doesn't necessarily include information about the module). Definitely agree that having 3 names on one function definition is a little much though.

Reply:
Having been the one to add DebugName, I think this is absolutely fine to use Dwarf when we can. It isn't stable anyway, so maybe rewrite the godoc and use it?

> DebugName returns the best non-empty name available for this function.
>
> For example, If DWARF is available, this will match the original source code name.
> Otherwise, this returns Name when non-empty or a generated name based on the
> Index.

Reply:
Sound great to me! I'll update.

Reply:
Done in https://github.com/tetratelabs/wazero/pull/1403/commits/5ab31ab98611fb6a2df3f62566f163fec501076a.